File: mockmate-backend/src/main/resources/runners/python/runner_string_array+int.py
import sys
import json
import re

def main():
    input_data = sys.stdin.read().strip()
    if not input_data:
        return

    start_idx = input_data.find('[')
    end_idx = input_data.rfind(']')

    if start_idx != -1 and end_idx != -1 and start_idx < end_idx:
        array_part = input_data[start_idx:end_idx+1]
        try:
            arr = json.loads(array_part.replace("'", '"'))
        except:
            arr = []
        remaining = input_data[end_idx+1:]
        nums = re.findall(r'-?\d+', remaining)
        k = int(nums[0]) if nums else 0
    else:
        arr, k = [], 0

    sol = Solution()
    result = sol.{{methodSignature}}(arr, k)

    if isinstance(result, bool):
        print(str(result).lower())
    else:
        print(json.dumps(result).replace(' ', ''))

{{USER_CODE}}

if __name__ == '__main__':
    main()

File: mockmate-backend/src/main/resources/runners/java/runner_string_array+int.java
import java.util.*;
import java.util.regex.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        while (sc.hasNextLine()) sb.append(sc.nextLine()).append(" ");
        String fullInput = sb.toString();

        try {
            int startBracket = fullInput.indexOf('[');
            int endBracket = fullInput.lastIndexOf(']');

            String[] arr;
            int k = 0;

            if (startBracket != -1 && endBracket != -1 && startBracket < endBracket) {
                String arrayPart = fullInput.substring(startBracket + 1, endBracket);
                List<String> list = new ArrayList<>();
                Matcher m = Pattern.compile("\"([^"]*)\"|'([^']*)'|([^\\s,]+)").matcher(arrayPart);
                while(m.find()) {
                    if(m.group(1) != null) list.add(m.group(1));
                    else if(m.group(2) != null) list.add(m.group(2));
                    else list.add(m.group(3));
                }
                arr = list.toArray(new String[0]);
                String remaining = fullInput.substring(endBracket + 1);
                Matcher mInt = Pattern.compile("-?\\d+").matcher(remaining);
                if(mInt.find()) k = Integer.parseInt(mInt.group());
            } else {
                arr = new String[0];
            }

            Solution sol = new Solution();
            Object result = sol.{{methodSignature}}(arr, k);
            System.out.println(result);
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        } finally {
            sc.close();
        }
    }
}

{{USER_CODE}}
